packages/alfa-selector/src/selector.ts

```diff
@@ -1249,6 +1249,8 @@ export namespace Selector {
       return new NthChild(index);
     }
 
+    private static readonly _indices = new WeakMap<Element, number>();
```

Comment:
Not sure why this is directly a weak map rather than a cache (which seems to be mostly a wrapper for weak map).

I kinda guess this is because it's filled in bulk (in `matches`) rather than one by one when a cache miss happens.
I would argue using a cache would make the intention clearer.

Reply:
`Cache` does provide some conveniences, but those come at a cost. Not a great cost, but a non-zero cost such as wrapping values in `Option`. As selector matching is one of our hottest paths, avoiding allocations like that is desirable and simple.

Reply:
Fair point 👍